actors/builtin/miner/partition_state_test.go

```diff
@@ -0,0 +1,591 @@
+		require.NoError(t, err)
+
+		// fault sector 3, 4, 5 and 6
+		faultSet := bf(3, 4, 5, 6)
+		faultSectors := selectSectors(t, sectors, faultSet)
+		_, err = partition.AddFaults(adt.AsStore(rt), faultSet, faultSectors, abi.ChainEpoch(7), sectorSize, quantSpec)
+		require.NoError(t, err)
+
+		// mark 4and 5 as a recoveries
+		recoverSet := bf(4, 5)
+		recoverSectors := selectSectors(t, sectors, recoverSet)
+		recoveredPower := miner.PowerForSectors(sectorSize, recoverSectors)
+		err = partition.AddRecoveries(recoverSet, recoveredPower)
+		require.NoError(t, err)
+
+		// now terminate 1, 3 and 5
+		terminations := bf(1, 3, 5)
+		terminatedSectors := selectSectors(t, sectors, terminations)
+		terminationEpoch := abi.ChainEpoch(3)
+		_, err = partition.TerminateSectors(adt.AsStore(rt), terminationEpoch, terminatedSectors, sectorSize, quantSpec)
+		require.NoError(t, err)
+
+		// pop first termination
+		result, hasMore, err := partition.PopEarlyTerminations(adt.AsStore(rt), 1)
+		require.NoError(t, err)
+
+		// expect first sector to be in early terminations
+		assertBitfieldsEqual(t, bf(1), result.Sectors[terminationEpoch])
+
+		// expect more results
+		assert.True(t, hasMore)
+
+		// expect terminations to still contain 3 and 5
+		queue, err := miner.LoadBitfieldQueue(adt.AsStore(rt), partition.EarlyTerminated, quantSpec)
+		require.NoError(t, err)
+
+		// only early termination appears in bitfield queue
+		ExpectBQ().
+			Add(terminationEpoch, 3, 5).
+			Equals(t, queue)
+
+		// pop the rest
+		result, hasMore, err = partition.PopEarlyTerminations(adt.AsStore(rt), 5)
+		require.NoError(t, err)
+
+		// expect 3 and 5
+		assertBitfieldsEqual(t, bf(3, 5), result.Sectors[terminationEpoch])
+
+		// expect no more results
+		assert.False(t, hasMore)
+
+		// expect early terminations to be empty
+		queue, err = miner.LoadBitfieldQueue(adt.AsStore(rt), partition.EarlyTerminated, quantSpec)
+		require.NoError(t, err)
+		ExpectBQ().Equals(t, queue)
+	})
+}
+
+type expectExpirationGroup struct {
+	expiration abi.ChainEpoch
+	sectors    *bitfield.BitField
+}
+
+func assertPartitionExpirationQueue(t *testing.T, rt *mock.Runtime, partition *miner.Partition, quant miner.QuantSpec, groups []expectExpirationGroup) {
+	queue, err := miner.LoadExpirationQueue(adt.AsStore(rt), partition.ExpirationsEpochs, quant)
+	require.NoError(t, err)
+
+	for _, group := range groups {
+		requireNoExpirationGroupsBefore(t, group.expiration, queue)
+		set, err := queue.PopUntil(group.expiration)
+		require.NoError(t, err)
+
+		// we pnly care whether the sectors are in the queue or not. ExpirationQueue tests can deal with early or on time.
+		allSectors, err := bitfield.MergeBitFields(set.OnTimeSectors, set.EarlySectors)
+		require.NoError(t, err)
+		assertBitfieldsEqual(t, group.sectors, allSectors)
+	}
+}
+
+func assertPartitionState(t *testing.T,
+	partition *miner.Partition,
+	sectors []*miner.SectorOnChainInfo,
```

Comment:
nit: I'd expect this to be _all_ sectors in the partition, terminated or otherwise.